connector/src/main/java/org/geysermc/connector/network/translators/java/entity/JavaEntityEquipmentTranslator.java

```diff
@@ -63,7 +66,24 @@ public void translate(GeyserSession session, ServerEntityEquipmentPacket packet)
             ItemData item = ItemTranslator.translateToBedrock(session, equipment.getItem());
             switch (equipment.getSlot()) {
                 case HELMET -> {
-                    livingEntity.setHelmet(item);
+                    CompoundTag profile = null;
```

Comment:
This can all be simplified:

```
if (livingEntity instanceof PlayerEntity
     && session.getItemMappings().getMapping(item).getJavaIdentifier().equals("minecraft:player_head")
     && equipment.getItem().getNbt() != null
     && equipment.getItem().getNbt().get("SkullOwner") instanceof CompoundTag profile) {
        FakeHeadProvider.setHead(session, (PlayerEntity) livingEntity, profile);
        ...
```